### ISSUE TITLE
Rate limiting

### ISSUE THREAD

**Issue by dct-infra** [LaneSh4d0w/query-streamlink#8]
Hey there!
Still on a trip?

I was just wondering about the techniques I could use to avoid rate limiting. ie: proxying  requests
I have been looking at the streamlink doc and read it was possible to use a specifc configuration file and pass several options through this config file.
I've tried but it didn't work for me, do you have any hints?
Thanks

**Comment by LaneSh4d0w** [comment]
> Hey there!
> Still on a trip?
> 
> I was just wondering about the techniques I could use to avoid rate limiting. ie: proxying  requests
> I have been looking at the streamlink doc and read it was possible to use a specifc configuration file and pass several options through this config file.
> I've tried but it didn't work for me, do you have any hints?
> Thanks

Hello there,

Yes I'm still abroad for work, I shall check the config documentation when I can.

**Comment by LaneSh4d0w** [comment]
Just thought about it now. On my code there's an extension called "Flask-Limiter". This can permit, using the remote IP of the end-user, to fix to each IP a certain number of accesses possible. If this limit is reached, it goes onto rate limit until the next minute.

**Comment by dct-infra** [comment]
That's exactly what I have disabled on the app side since in our case everything incoming goes masqueraded through the app firewall. Only app fw IP visible to the app.

But i'm interested in knowing how the limits work
#@limiter.limit("20/minute")
#@limiter.limit("1/second")
I assume the first line allows a single IP to establish 20 connections per minute total with a rate of max 1 per second?
Please correct if I'm wrong.

**Comment by LaneSh4d0w** [comment]
> 
> 
> That's exactly what I have disabled on the app side since in our case everything incoming goes masqueraded through the app firewall. Only app fw IP visible to the app.
> 
> But i'm interested in knowing how the limits work #@limiter.limit("20/minute") #@limiter.limit("1/second") I assume the first line allows a single IP to establish 20 connections per minute total with a rate of max 1 per second? Please correct if I'm wrong.

You are right. Limiter will rate limit users to 1 request / second and up to 20 requests per minute.
